pkg/networkpath/traceroute/common/common_windows.go

```diff
@@ -0,0 +1,184 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package common contains common functionality for both TCP and UDP
+// traceroute implementations
+package common
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"golang.org/x/net/ipv4"
+	"golang.org/x/sys/windows"
+)
+
+var (
+	sendTo   = windows.Sendto
+	recvFrom = windows.Recvfrom
+)
+
+type (
+	// Winrawsocket is a struct that encapsulates a raw socket
+	// on Windows that can be used to listen to traffic on a host
+	// or send raw packets from a host
+	Winrawsocket struct {
```

Comment:
Do we not have an implementation in the ipv4 package for windows?

Reply:
Got it. Could this be made more like the other raw socket implementations in the go libraries? Is there some interface we can implement that would make its use look more familiar for people who have used the networking go libraries?

Reply:
Do you have any specific thoughts here? In the Linux implementation it's already kind of clunky because in Go we have to go from a `net.ListenPacket` then to get a raw socket we have to take that connection and pass it into `ipv4.NewRawConn`.

`sendTo` and `readFrom` are the same names as in the Linux version so I think those should be okay

Maybe I'll re-name the `CreateRawSocket()` to `NewWindowsRawConn`?